client/components/product-details.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {}
```

Comment:
If you send your initial product state to `null`, then you can conditionally render your react elements.

```js
if (this.state.product === null) return null
return (
  // react elements
)
```